### PR TITLE
ci-secret-bootstrap: --secret-names

### PR DESCRIPTION
**Pull request by hongkailiu**
https://issues.redhat.com/browse/DPTP-2585

/cc @openshift/test-platform

### CODE REVIEW

**Comment by smg247** [comment]
Do you think there would be any future value to making this argument more generic? Something like, `secretsNamed` and then pass a value (in this case `config-updater`). That way we could use this for other cases of only working with specifically named secrets.

File: cmd/ci-secret-bootstrap/main.go [hunk 1]
```diff
@@ -120,6 +122,12 @@ func (o *options) completeOptions(censor *secrets.DynamicCensor, kubeConfigs map
 		return err
 	}
 
+	if o.configUpdaterOnly {
+		logrus.WithField("o.configUpdaterOnly", o.configUpdaterOnly).Info("Removing secrets whose name is not config-updater ...")
```
Comment (smg247) [inline cmd/ci-secret-bootstrap/main.go:126]: The field on this log line will only ever be true. I think a regular old `logrus.Info` would suffice here.

Comment (hongkailiu) [reply cmd/ci-secret-bootstrap/main.go]: Updated.

File: cmd/ci-secret-bootstrap/main.go [hunk 2]
```diff
@@ -120,6 +122,12 @@ func (o *options) completeOptions(censor *secrets.DynamicCensor, kubeConfigs map
...
+		configUpdaterOnly(&o.config)
+		logrus.WithField("o.config.Secrets", o.config.Secrets).WithField("o.configUpdaterOnly", o.configUpdaterOnly).Info("Removed secrets whose name is not config-updater")
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/ci-secret-bootstrap/main.go]: Updated.

File: cmd/ci-secret-bootstrap/main.go [hunk 3]
```diff
@@ -120,6 +122,13 @@ func (o *options) completeOptions(censor *secrets.DynamicCensor, kubeConfigs map
 		return err
 	}
 
+	if vals := o.secretNamesRaw.Strings(); len(vals) > 0 {
+		secretNames := sets.NewString(vals...)
+		logrus.Info("Removing secrets whose name is not config-updater ...")
+		configUpdaterOnly(&o.config, secretNames)
```
Comment (smg247) [inline cmd/ci-secret-bootstrap/main.go:128]: This method and preceeding log call should also be genericized

File: cmd/ci-secret-bootstrap/main_test.go [hunk 4]
```diff
@@ -2830,3 +2830,59 @@ func TestIntegration(t *testing.T) {
...
+					},
+				},
+			},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			configUpdaterOnly(tc.given, sets.NewString("config-updater"))
+			if diff := cmp.Diff(tc.given, tc.expected); diff != "" {
+				t.Errorf("%s: actual differs from expected after casin update e: %s", tc.name, diff)
```
Comment (smg247) [inline cmd/ci-secret-bootstrap/main_test.go:2884]: "casin"? typo?

Comment (hongkailiu) [reply cmd/ci-secret-bootstrap/main_test.go]: Leftover from copying.
Removed.

File: cmd/ci-secret-bootstrap/main.go (hunk 3 above)
Comment (hongkailiu) [reply cmd/ci-secret-bootstrap/main.go]: Updated.

**Review by smg247 — APPROVED** [review]

**Pull request merged**